scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/ServerHandshaker.java

```diff
@@ -261,17 +261,16 @@ protected void doProcessMessage(HandshakeMessage message) throws HandshakeExcept
 	private void receivedClientCertificate(final CertificateMessage message) throws HandshakeException {
 
 		clientCertificate = message;
+		clientPublicKey = message.getPublicKey();
 		if (clientAuthenticationRequired && message.getCertificateChain() != null
-				&& message.getPublicKey() == null) {
+				&& clientPublicKey == null) {
```

Comment:
I suppose this is a kind of optimisation ? 

If it is, maybe we should go to execute `message.getPublicKey()` only `if clientAuthenticationRequired && message.getCertificateChain() != null` is `true`

```java
clientPublicKey = null; // I'm not sure this must be initialized to null
if (clientAuthenticationRequired && message.getCertificateChain() != null){
    clientPublicKey = message.getPublicKey();
    if  (clientPublicKey == null) {
```

But looking at this it seems that verifyCertificate will call message.getPublicKey() too... :thinking: 

So, maybe I'm totally wrong and the purpose of this is not "an optimization" at all ? :sweat_smile:

Reply:
Ok I looked at #1145 and this PR is clearly not an "optimization".

So the question is what is its purpose :sweat_smile:  ?

Reply:
I implemented that before the other PR.
FMPOV, calling get multiple times expecting the same value is somehow smelly.

Reply:
for a getter this does not disturb me a lot

Reply:
Depends on expecting the same value or updated values.